gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/FeatureRepositionViewModel.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.ui.home.mapcontainer;
+
+import com.google.android.gnd.model.feature.Point;
+import com.google.android.gnd.rx.Nil;
+import com.google.android.gnd.rx.annotations.Hot;
+import com.google.android.gnd.ui.common.AbstractViewModel;
+import com.google.android.gnd.ui.common.SharedViewModel;
+import com.google.android.gnd.ui.map.MapProvider;
+import io.reactivex.Observable;
+import io.reactivex.subjects.PublishSubject;
+import io.reactivex.subjects.Subject;
+import javax.inject.Inject;
+
+@SharedViewModel
+public class FeatureRepositionViewModel extends AbstractViewModel {
+
+  private final MapProvider mapProvider;
+
+  @Hot private final Subject<Point> confirmButtonClicks = PublishSubject.create();
+  @Hot private final Subject<Nil> cancelButtonClicks = PublishSubject.create();
+
+  @Inject
+  FeatureRepositionViewModel(MapProvider mapProvider) {
```

Comment:
@gino-m Should we directly use `MapProvider` or use `MapRepository`?

Reply:
Feels like there should be a separate abstraction which should provide location updates rather than exposing `getCameraTarget` indirectly all the way to the top.
`GoogleMapsAdapter` -> `GoogleMapsProvider` -> `VM`

Wdyt?

Reply:
This is definitely wrong.. ViewModels should never reference Views since the lifecycle of VMs is longer. Even though MapProvider isn't a view class, it returns a the map Fragment which is. I think the right thing to do here would be for `FeatureRepositionView` to subscribe to camera movements, calling a setter on `FeatureRepositionViewModel` on each change to update the current VM state. Wdyt?

Reply:
Done